src/test/ui/closures/multiple-fn-bounds.stderr

```diff
@@ -0,0 +1,24 @@
+error[E0631]: type mismatch in closure arguments
+  --> $DIR/multiple-fn-bounds.rs:10:5
+   |
+LL |     foo(move |x| v);
+   |     ^^^ -------- found signature defined here
+   |     |
+   |     expected due to this
+   |
+   = note: expected closure signature `fn(char) -> _`
+              found closure signature `for<'r> fn(&'r char) -> _`
+note: closure inferred to have a different signature due to this bound
```

Comment:
Ideally this would render below this other note below, but that note is added way later in the process, so it's kinda impossible to do without really refactoring stuff I think.